src/Framework/Traits.cs

```diff
@@ -30,7 +30,7 @@ public Traits()
         {
             EscapeHatches = new EscapeHatches();
 
-            DebugScheduler = DebugEngine || !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDDEBUGSCHEDULER"));
+            DebugScheduler = !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("MSBUILDDEBUGSCHEDULER"));
```

Comment:
Let's update the doc accordingly

https://github.com/dotnet/msbuild/blob/5e9db6f069b2a3fab5d651b7273731413473b673/documentation/wiki/MSBuild-Environment-Variables.md?plain=1#L29